kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -540,4 +544,143 @@ class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase
       }
     }
   }
+
+  test("JsonMetadataDomain can serialize/deserialize Optional fields") {
```

Comment:
It looks like this does not depend on the engine. Can you move tests like this to the `kernel-api` module?

Reply:
Sure, I've moved tests that don't depend on the engine to the `kernel-api`.